CMakeLists.txt

```diff
@@ -123,7 +123,10 @@ endif ()
 
 set_target_properties("Zycore" PROPERTIES LINKER_LANGUAGE C)
 target_include_directories("Zycore"
-    PUBLIC "include" ${PROJECT_BINARY_DIR}
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/Zycore>
```

Comment:
@abouvier I'm not sure that adding `${CMAKE_INSTALL_INCLUDEDIR}/Zycore` to the list of PUBLIC include directories is needed, since users should include headers specifying the Zycore directory (`#include <Zycore/Allocator.h>` instead of `#include <Allocator.h>`, see the examples)
https://github.com/zyantific/zycore-c/blob/22ce9c2de5c5dbc0d09077377695fdbc7ad5fe64/examples/String.c#L33

Reply:
It's what CMake do in their [own documentation](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html).
And by looking to the *.cmake files installed on my system, virtually all of them do the same.

Reply:
[SDL2 use both paths](https://github.com/libsdl-org/SDL/commit/1d624aa5730adcccf846267de76b13285bc65fed) for example.
But I don't know in the end, I'm not really a CMake expert :p

Reply:
Hmm, yeah, I tend to agree that this **shouldn't** be required. Didn't notice it during initial review, thanks for pointing it out @Tachi107! Perhaps SDL2 etc need this because internal includes don't use fully qualified `<SDL2/xxx.h>` includes internally?